flang/lib/Lower/ConvertCall.cpp

```diff
@@ -1172,8 +1173,6 @@ genUserCall(Fortran::lower::PreparedActualArguments &loweredActuals,
       continue;
     }
     const auto *expr = arg.entity->UnwrapExpr();
```

Comment:
If you apply my comment above, you can move this to the MutableBox case where it is used with an assert that this is not null (it is not possible to pass an assumed type to a pointer/allocatable).